### PR TITLE
Prepare docs for Aug 2023 3rd wave of Providers (RC2)

### PR DESCRIPTION
**Pull request by eladkal**
Note: Due to the fact that there were many commits added to azure provider the between RC1 to RC2 I will not use shorten vote period. The vote will last for 72 hours.

### CODE REVIEW

File: airflow/providers/cncf/kubernetes/CHANGELOG.rst [hunk 1]
```diff
@@ -44,6 +44,7 @@ Bug Fixes
 * ``Fix KubernetesPodOperator duplicating logs when interrupted (#33500)``
 * ``Fix 2.7.0 db migration job errors (#33652)``
 * ``Inspect container state rather than last_state when deciding whether to skip (#33702)``
+* ``Update pod.py (#33779)``
```
Comment (eladkal) [inline airflow/providers/cncf/kubernetes/CHANGELOG.rst]: FYI in providers we can not update commit messages in release notes

**Review by potiuk — APPROVED** [review]

**Review by hussein-awala — APPROVED** [review]

File: airflow/providers/cncf/kubernetes/CHANGELOG.rst (hunk 1 above)
Comment (potiuk) [reply airflow/providers/cncf/kubernetes/CHANGELOG.rst]: @eladkal -> yes, we can update the message in CHANGELOG - we cannot update them in COMMITS. The only thing important is to make sure we leave `(#33779)`.

Comment (eladkal) [reply airflow/providers/cncf/kubernetes/CHANGELOG.rst]: I tried that once and it didn't work(?) let me try again

**Review by potiuk — APPROVED** [review]

File: airflow/providers/microsoft/azure/CHANGELOG.rst [hunk 2]
```diff
@@ -36,12 +36,19 @@ Features
...
+* ``Allow passing fully_qualified_namespace and credential to initialize Azure Service Bus Client (#33493)``
+* ``Add DefaultAzureCredential support to cosmos (#33436)``
+* ``Add DefaultAzureCredential support to AzureBatchHook (#33469)``
 
 Bug Fixes
 ~~~~~~~~~
 
 * ``Fix updating account url for WasbHook (#33457)``
 * ``Fix Azure Batch Hook instantation (#33731)``
+* ``Truncate Wasb storage account name if it's more than 24 characters (#33851)``
```
Comment (ephraimbuddy) [inline airflow/providers/microsoft/azure/CHANGELOG.rst:49]: We can combine this with line 47

**Review by ephraimbuddy — APPROVED** [review]

File: airflow/providers/cncf/kubernetes/CHANGELOG.rst (hunk 1 above)
Comment (potiuk) [reply airflow/providers/cncf/kubernetes/CHANGELOG.rst]: It was probabaly in "COMMITS" :). The COMMITS list is always regenerated from scratch when the docs are generated - just to make sure it is accurate. The CHANGELOG on the other hand is only updated with new things (based on the 'PR#') and we only add new things to changelog if we see the #PR  number is missing there. This allows to manually modify and move around the individual entries - hide them, reword them, etc.

Comment (potiuk) [reply airflow/providers/cncf/kubernetes/CHANGELOG.rst]: Also this allows to make "iterations" of release. Whenever we "prepare docs" - like this one from the new main, only the "new" things that had not been present the last time in the CHANGELOG are added. That's why when you cancel an RC and prepare the new one, all those "new entries" land in the "commented" section stating "Please update and move the new entries that appeared since the last "prepare documantation" - that effectively allows to prepare same version several times (several RC candidates or even several times before deciding on RC). The (PR#) is the key to see if new entries should be added when you generate the docs from new main.

File: airflow/providers/microsoft/azure/CHANGELOG.rst (hunk 2 above)
Comment (eladkal) [reply airflow/providers/microsoft/azure/CHANGELOG.rst:49]: I will try to sort it after the release. There can be several such improvements I'll handle them in bulk
